recipes/ms-gsl/all/conanfile.py

```diff
@@ -50,8 +57,14 @@ def package_id(self):
         self.info.header_only()
 
     def package_info(self):
+        if tools.Version(self.version) >= "3.0.0":
+            if self.options.on_contract_violation != 'terminate':
+                raise ConanInvalidConfiguration("Microsoft GSL only supports terminate on contract violation since version 3.0.0")
+
         self.cpp_info.defines = [
             self._contract_map[str(self.options.on_contract_violation)]
         ]
-        self.cpp_info.names["cmake_find_package"] = "Microsoft.GSL"
-        self.cpp_info.names["cmake_find_package_multi"] = "Microsoft.GSL"
+
+        if tools.Version(self.version) >= "3.0.0":
+            self.cpp_info.names["cmake_find_package"] = "Microsoft.GSL"
+            self.cpp_info.names["cmake_find_package_multi"] = "Microsoft.GSL"
```

Comment:
```suggestion
            self.cpp_info.names["cmake_find_package_multi"] = "Microsoft.GSL"
        else:
            self.cpp_info.names["cmake_find_package"] = "GSL"
            self.cpp_info.names["cmake_find_package_multi"] = "GSL"
```

https://gitlab.kitware.com/cmake/cmake/-/blob/master/Modules/FindGSL.cmake

Reply:
Find the native GNU Scientific Library (GSL) includes and libraries.

Reply:
https://github.com/microsoft/GSL/issues/31